panther_driver/src/battery_node.py

```diff
@@ -2,153 +2,131 @@
-                #     f'I_bat2={I_bat2:.2f}, V_bat2={V_bat2:.2f}, Idriv2={Idriv2:.2f}, V_driv2={V_driv2:.2f}, I_charge_bat2={I_charge_bat2:.2f}, temp_bat2={temp_bat2:.2f}')
-
-                self._publish_battery_msg(
-                    self._battery1_publisher, True, V_bat1, temp_bat1, -I_bat1 + I_charge_bat1
-                )
-                self._publish_battery_msg(
-                    self._battery2_publisher, True, V_bat2, temp_bat2, -I_bat2 + I_charge_bat2
-                )
-
-                V_bat_avereage = (V_bat1 + V_bat2) / 2
-                temp_average = (temp_bat1 + temp_bat2) / 2
-                I_bat_average = (I_bat1 + I_bat2) / 2
-                I_charge_bat_average = (I_charge_bat1 + I_charge_bat2) / 2
-
-                self._publish_battery_msg(
-                    self._battery_publisher,
-                    True,
-                    V_bat_avereage,
-                    temp_average,
-                    -I_bat_average + I_charge_bat_average,
-                )
-        except:
-            rospy.logerr(f'[{rospy.get_name()}] Error Calculating and publishing bat data')
+        # Check battery num
+        if V_temp_bat2 > 3.03:  # One battery
+            # Calculate Temp in deg of Celcius
+            temp_bat1 = self._voltage_to_deg(V_temp_bat1)
+
+            self._publish_battery_msg(self._battery1_publisher, True, V_bat1, temp_bat1, I_bat1)
+            self._publish_battery_msg(self._battery2_publisher, False)
+        
+        else:
+            # Calculate Temp in deg of Celcius
+            temp_bat1 = self._voltage_to_deg(V_temp_bat1)
+            temp_bat2 = self._voltage_to_deg(V_temp_bat2)
+
+            self._publish_battery_msg(
+                self._battery1_publisher, True, V_bat1, temp_bat1, -I_bat1 + I_charge_bat1
+            )
+            self._publish_battery_msg(
+                self._battery2_publisher, True, V_bat2, temp_bat2, -I_bat2 + I_charge_bat2
+            )
+
+            V_bat_avereage = (V_bat1 + V_bat2) / 2.0
+            temp_average = (temp_bat1 + temp_bat2) / 2.0
+            I_bat_average = (I_bat1 + I_bat2) / 2.0
+            I_charge_bat_average = (I_charge_bat1 + I_charge_bat2) / 2.0
 
-    def _get_ADC_measurement(self, name: str, config_file):
+            self._publish_battery_msg(
+                self._battery_publisher, True, V_bat_avereage, temp_average, -I_bat_average + I_charge_bat_average
+            )
+
+    def _voltage_to_deg(self, V_temp) -> float:
+        if V_temp == 0 or V_temp >= self._U_supply:
+            rospy.logerr(f'[{rospy.get_name()}] Temperature measurement error')
+            return float('nan')
+
+        R_therm = (V_temp * self._R1) / (self._U_supply - V_temp)
+
+        return (self._A * self._B / (self._A * math.log(R_therm / self._R0) + self._B)) - 273.15
```

Comment:
Does it make sense to make this static method? `_A _B _R1 _R0 _U_supply` are only used inside it.